### DISCUSSION TITLE
Compatability with Illandril's Character Class Enhancements (5e)

### DISCUSSION THREAD

**Discussion by KikinChikin** [misthero/dnd5e-spellpoints#101] [Ideas]
Both modules are very cool, but while trying to give a 3/4 caster spell points they wont receive more than 1. It would honestly be really dope if they worked together as both expand casters and allow for more flexibility and diversity among casters.

**Comment by misthero** [comment]
can you get to explain more in detail what is happening? I'm not very familiar with Illandril's Character Class Enhancements.

**Reply by KikinChikin** [reply]
See im not 100% sure what going on in the background. But Illandril's Character Class Enhancements adds a 3/4 caster, so at level 20 they are effectively a level 15 caster and when i apply that caster type to a class, the calculation to determine how many spell points they have seems to fail and leaves them with 1 spell point.

Sorry this reply took so long, i was experimenting and trying to fix the issue or at least identify where things go wrong.

**Comment by misthero** [comment]
Ok I got why it is not working, the spellpoints module looks in the class item to check the class spellcasting progression so it knows that the character is a spellcaster and what kind of spellcaster it is. It specifically looks for the standard progressions that in 5e are the following:

- full
- half
- third
- artificier
- pact

depending on the progression it does the calculations following the Dungeon Master Guide.

but that module, the Character Class Enhancements changes that identifier, I see in my test "illandrilTwoThirds"

![immagine](https://github.com/user-attachments/assets/d4b45ca3-faea-409e-8f5a-ab609f69113d)

so the spellpoints module is not able to determine that is a spellcaster and how to calculate it.

I will investigate more to see if a workaround is possible, but for now I would say that the automatic calculations is not compatible with that module, you have to manually change the spellpoints max until I find a solution or develop a compatibility layer specifically for that module.

![immagine](https://github.com/user-attachments/assets/2994cc97-6b5b-45de-b176-bed5cb704648)


Sorry.

**Reply by KikinChikin** [reply]
Alrighty, thank you. I am not running my game yet so ill let you work and hope there is a work around before i start running.
